src/log.cpp

```diff
@@ -187,13 +190,48 @@ void set_log_to_file()
 	if(is_log_dir_writable_.value_or(false)) {
 		// get the log file stream and assign cerr+cout to it
 		output_file_path_ = filesystem::get_logs_dir()+"/"+unique_log_filename();
-		static std::unique_ptr<std::ostream> logfile { filesystem::ostream_file(output_file_path_) };
-		static std::ostream cerr_stream{std::cerr.rdbuf()};
-		//static std::ostream cout_stream{std::cout.rdbuf()};
-		auto cerr_tee { boost::iostreams::tee(*logfile, cerr_stream) };
-		output_file_.reset(new boost::iostreams::stream<decltype(cerr_tee)>{cerr_tee, 4096, 0});
-		std::cerr.rdbuf(output_file_.get()->rdbuf());
-		std::cout.rdbuf(output_file_.get()->rdbuf());
+
+		// redirect to a file
+		// IMPORTANT: apparently redirecting stderr/stdout will also redirect std::cerr/std::cout, but the reverse is not true
+		//            redirecting std::cerr/std::cout will *not* redirect stderr/stdout
+		if(setvbuf(stdout, NULL, _IONBF, 2) == -1) {
```

Comment:
Why do you want to set stdout to be unbuffered?

Reply:
Otherwise the stdout/std::cout output doesn't make it to the log file, at least for the example in wesnoth.cpp where it almost immediately terminates with an assert.

Reply:
Hmm… std::cout and/or the C FILE API might have a software buffer on top of that, so maybe it's fine…